### ISSUE TITLE
Pundit authorization doesn't authorize index view

### ISSUE THREAD

**Issue by niedfelj** [thoughtbot/administrate#1822]
* What were you trying to do?
Secure the admin dashboard using Pundit with include Administrate::Punditize

* What did you end up with (logs, or, even better, example apps are great!)?
It seems that the index action does not call authorize_resource

* What versions are you running?
  - Rails 6.0.3.4
  - administrate 0.14.0

I'm confused why administrate is not attempting to call authorize_resource in index calls? I've had to monkey patch my admin controller with the following to properly prevent any user from access the resource index. Is the expectation that limited scopes should be used for this instead? 

```
    def index 
      authorize resource_class
      super 
    end
```

**Comment by pablobm** [comment]
Thank you for raising this. It does sound like incorrect behaviour: it should be possible to block access to index pages altogether, just limit the listings with a scope.

Do you think you could contribute a PR to fix this?

**Comment by nickcharlton** [comment]
There's an existing issue here, that's been around a little while on this: #1233 and a PR too: #1234.

**Comment by niedfelj** [comment]
@nickcharlton @pablobm That PR is exactly what I did to solve it.

**Comment by nickcharlton** [comment]
Ah great, thanks! I'll find some time to take a look at that today. It's been on my short-list of issues, but I wasn't quite sure if it was still a problem.

**Comment by nickcharlton** [comment]
Thanks for reporting this! I've updated and merged #1234.
